src/stflstring.cpp

```diff
@@ -0,0 +1,153 @@
+#include "stflstring.h"
+
+#include <utility>
+
+#include "utils.h"
+
+namespace newsboat {
+
+StflString::StflString(std::string&& text, std::map<size_t, std::string>&& style_tags)
+	: text(text)
+	, style_tags(style_tags)
```

Comment:
This doesn't perform any moves. I don't have a satisfactory explanation of *why* it doesn't, but you can try to write the same code for move-only objects and see that it doesn't compile: https://godbolt.org/z/4T1as91P6 You have to wrap arguments into `std::move`, even though they are already rvalues.

(I know this PR is just a draft for now, but I couldn't resist taking a peek. Good work!)

Reply:
Good catch, resolved in https://github.com/newsboat/newsboat/pull/2084/commits/4add925489e16d0636082cd3b6e30418380f4431